tests/platform_tests/api/test_sfp.py

```diff
@@ -151,6 +151,136 @@ class TestSfpApi(PlatformApiTestBase):
+        'lasertemphighalarm'
+    ]
+
+    QSFPZR_EXPECTED_XCVR_THRESHOLD_INFO_KEYS = [
+        'txpowerlowwarning',
+        'temphighwarning',
+        'temphighalarm',
+        'txbiashighalarm',
+        'vcchighalarm',
+        'txbiaslowalarm',
+        'rxpowerhighwarning',
+        'vcclowwarning',
+        'txbiaslowwarning',
+        'rxpowerlowalarm',
+        'vcchighwarning',
+        'txpowerhighwarning',
+        'rxpowerlowwarning',
+        'txbiashighwarning',
+        'vcclowalarm',
+        'txpowerhighalarm',
+        'templowalarm',
+        'rxpowerhighalarm',
+        'templowwarning',
+        'txpowerlowalarm',
+        'lasertemphighwarning',
+        'lasertemplowwarning',
+        'lasertemplowalarm',
+        'lasertemphighalarm',
+        'prefecberhighalarm',
+        'prefecberlowalarm',
+        'prefecberhighwarning',
+        'prefecberlowwarning',
+        'postfecberhighalarm',
+        'postfecberlowalarm',
+        'postfecberhighwarning',
+        'postfecberlowwarning',
+        'biasxihighalarm',
+        'biasxilowalarm',
+        'biasxihighwarning',
+        'biasxilowwarning',
+        'biasxqhighalarm',
+        'biasxqlowalarm',
+        'biasxqhighwarning',
+        'biasxqlowwarning',
+        'biasxphighalarm',
+        'biasxplowalarm',
+        'biasxphighwarning',
+        'biasxplowwarning',
+        'biasyihighalarm',
+        'biasyilowalarm',
+        'biasyihighwarning',
+        'biasyilowwarning',
+        'biasyqhighalarm',
+        'biasyqlowalarm',
+        'biasyqhighwarning',
+        'biasyqlowwarning',
+        'biasyphighalarm',
+        'biasyplowalarm',
+        'biasyphighwarning',
+        'biasyplowwarning',
+        'cdshorthighalarm',
+        'cdshortlowalarm',
+        'cdshorthighwarning',
+        'cdshortlowwarning',
+        'cdlonghighalarm',
+        'cdlonglowalarm',
+        'cdlonghighwarning',
+        'cdlonglowwarning',
+        'dgdhighalarm',
+        'dgdlowalarm',
+        'dgdhighwarning',
+        'dgdlowwarning',
+        'sopmdhighalarm',
+        'sopmdlowalarm',
+        'sopmdhighwarning',
+        'sopmdlowwarning',
+        'pdlhighalarm',
+        'pdllowalarm',
+        'pdlhighwarning',
+        'pdllowwarning',
+        'osnrhighalarm',
+        'osnrlowalarm',
+        'osnrhighwarning',
+        'osnrlowwarning',
+        'esnrhighalarm',
+        'esnrlowalarm',
+        'esnrhighwarning',
+        'esnrlowwarning',
+        'cfohighalarm',
+        'cfolowalarm',
+        'cfohighwarning',
+        'cfolowwarning',
+        'txcurrpowerhighalarm',
+        'txcurrpowerlowalarm',
+        'txcurrpowerhighwarning',
+        'txcurrpowerlowwarning',
+        'rxtotpowerhighalarm',
+        'rxtotpowerlowalarm',
+        'rxtotpowerhighwarning',
+        'rxtotpowerlowwarning',
+        'rxsigpowerhighalarm',
+        'rxsigpowerlowalarm',
+        'rxsigpowerhighwarning',
+        'rxsigpowerlowwarning'
+    ]
+
```

Comment:
Seems like  QSFPDD table  = XCVR table + new fields

and 

QSFPZR table =  QSFPDD table + new fields.

Might make sense to have only the new fields in QSPFDD and QSPFZR tables.

So, code below might look like:

```
expected_keys = self.EXPECTED_XCVR_THRESHOLD_INFO_KEYS
if info_dict["type_abbrv_name"] == "QSFP-DD":
  expected_keys += self.QSFPDD_EXPECTED_XCVR_THRESHOLD_INFO_KEYS
  if 'ZR' in info_dict["media_interface_code"]:
     expected_keys += self.QSFPZR_EXPECTED_XCVR_THRESHOLD_INFO_KEYS
```

Reply:
agree with @mannytaheri. @rajann can you make this change?

Reply:
Sure..I can make this change.  Hoping this list always backward compatible.